### PR TITLE
Replace json.Unmarshal() with json.Decoder().Decode()

### PR DESCRIPTION
**Pull request by runcom**
Signed-off-by: Antonio Murdaca me@runcom.ninja

### CODE REVIEW

**Comment by LK4D4** [comment]
LGTM

File: builder/parser/line_parsers.go [hunk 1]
```diff
@@ -233,7 +233,7 @@ func parseString(rest string) (*Node, map[string]bool, error) {
 // parseJSON converts JSON arrays to an AST.
 func parseJSON(rest string) (*Node, map[string]bool, error) {
 	var myJson []interface{}
-	if err := json.Unmarshal([]byte(rest), &myJson); err != nil {
+	if err := json.NewDecoder(strings.NewReader(rest)).Decode(&myJson); err != nil {
```
Comment (tiborvass) [inline builder/parser/line_parsers.go]: there's no need for this. It's better to use an io.Reader only when the content is not all in memory. In this case it's already in memory so I don't see why not keep Unmarshal.

File: pkg/parsers/filters/parse.go [hunk 2]
```diff
@@ -58,8 +58,7 @@ func FromParam(p string) (Args, error) {
 	if len(p) == 0 {
 		return args, nil
 	}
-	err := json.Unmarshal([]byte(p), &args)
-	if err != nil {
+	if err := json.NewDecoder(strings.NewReader(p)).Decode(&args); err != nil {
```
Comment (tiborvass) [inline pkg/parsers/filters/parse.go]: same here, we can keep Unmarshal.

File: builder/parser/line_parsers.go (hunk 1 above)
Comment (LK4D4) [inline builder/parser/line_parsers.go]: Actually it isn't an issue. `json.Decode` unmarshal whole content from `Reader` anyway if we have only one object(90% of cases in docker) and it is same by memory consumption. But `Decoder` uses internal `sync.Pool` for storing some data structures in cache.

**Comment by tiborvass** [comment]
I'll give in to @LK4D4 :) LGTM

**Pull request merged**